planner/core/common_plans.go

```diff
@@ -385,25 +401,69 @@ func (e *Execute) rebuildRange(p Plan) error {
 		if err != nil {
 			return err
 		}
+		if is.Table.Partition != nil {
+			pi := is.Table.Partition
+			if pi.Type == model.PartitionTypeHash && len(is.Ranges) == 1 && is.Ranges[0].IsPoint(sc) {
+				schema, names := buildSchemaAndNameFromIndex(is.IdxCols, is.DBName, is.Table, is.Index)
+				expr, err := expression.ParseSimpleExprsWithNames(e.ctx, pi.Expr, schema, names)
+				if err != nil {
+					return err
+				}
+				pos, err := locateHashPartition(e.ctx, expr[0], pi, is.Ranges[0].LowVal)
+				if err != nil {
+					return err
+				}
+				pID := pi.Definitions[pos].ID
+				is.physicalTableID = pID
+			}
+		}
 	case *PhysicalIndexLookUpReader:
 		is := x.IndexPlans[0].(*PhysicalIndexScan)
 		is.Ranges, err = e.buildRangeForIndexScan(sctx, is)
 		if err != nil {
 			return err
 		}
+		if is.Table.Partition != nil {
+			pi := is.Table.Partition
+			if pi.Type == model.PartitionTypeHash && len(is.Ranges) == 1 && is.Ranges[0].IsPoint(sc) {
+				schema, names := buildSchemaAndNameFromIndex(is.IdxCols, is.DBName, is.Table, is.Index)
+				expr, err := expression.ParseSimpleExprsWithNames(e.ctx, pi.Expr, schema, names)
+				if err != nil {
+					return err
+				}
+				pos, err := locateHashPartition(e.ctx, expr[0], pi, is.Ranges[0].LowVal)
+				if err != nil {
+					return err
+				}
+				pID := pi.Definitions[pos].ID
+				is.physicalTableID = pID
+				tblScan := x.TablePlans[0].(*PhysicalTableScan)
+				tblScan.physicalTableID = pID
+			}
+		}
 	case *PointGetPlan:
 		if x.HandleParam != nil {
 			x.Handle, err = x.HandleParam.Datum.ToInt64(sc)
 			if err != nil {
 				return err
 			}
+			if x.PartitionInfo != nil {
+				num := x.TblInfo.Partition.Num
+				pos := x.Handle % int64(num)
+				x.PartitionInfo = &x.TblInfo.Partition.Definitions[pos]
+			}
 			return nil
 		}
 		for i, param := range x.IndexValueParams {
 			if param != nil {
 				x.IndexValues[i] = param.Datum
 			}
 		}
+		if x.PartitionInfo != nil {
+			val := x.IndexValues[x.partitionColumnPos].GetInt64()
+			partitionID := val % int64(x.TblInfo.Partition.Num)
+			x.PartitionInfo = &x.TblInfo.Partition.Definitions[partitionID]
+		}
 		return nil
 	case *BatchPointGetPlan:
```

Comment:
Shall `BatchPointGetPlan` be handled as well?

Reply:
Table partition does not support `BatchPointGet` yet.